packages/js/uri-resolvers/src/helpers/RetryResolver.ts

```diff
@@ -0,0 +1,113 @@
+import { UriResolverLike } from "./UriResolverLike";
+import { UriResolver } from "./UriResolver";
+import { UriResolutionResult } from "./UriResolutionResult";
+
+import {
+  CoreClient,
+  IUriResolutionContext,
+  IUriResolver,
+  Result,
+  Uri,
+  UriPackageOrWrapper,
+} from "@polywrap/core-js";
+
+/** configure how additional resolution attempts are handled after an initial resolution attempt fails */
+export type RetryOptions = {
+  /** the number of additional resolution attempts */
+  retries: number;
+  /** the duration (in ms) to pause between attempts */
+  interval: number;
```

Comment:
A good practice for time variables is to put the unit in the name. E.g. intervalMs/intervalInMilliseconds/intervalMilliseconds